Scout test cases for Security

enriquesanchez-elastic opened the pull request:
## Overview

This PR migrates the Security Solution AI Assistant test suite from Cypress to Scout (Playwright), establishing a reference implementation for other Security Solution teams to follow. The migration includes comprehensive test coverage, improved architecture patterns, and extensive documentation.

## 📊 Migration Scope

| Metric | Value |
|--------|-------|
| **Cypress test files migrated** | 7 files → 6 Scout spec files |
| **Test scenarios** | 29 comprehensive test cases |
| **New page objects** | 4 classes (~1,300 lines) |
| **New API services** | 3 services (~750 lines) |
| **Test infrastructure improvements** | 8 critical fixes applied |

## 🎯 Key Accomplishments

### 1. Complete AI Assistant Test Coverage

**Test Files Migrated:**
- ✅ `feature_availability.spec.ts` - License tier and feature availability (3 test suites)
- ✅ `conversations.spec.ts` - Core conversation management (9 tests)
- ✅ `prompts.spec.ts` - System and quick prompt management (7 tests)
- ✅ `messages.spec.ts` - Message rendering and timeline integration (1 test)
- ✅ `shared_conversations.spec.ts` - Multi-user conversation sharing (9 tests, currently skipped)
- ✅ `security_navigation.spec.ts` - Security navigation patterns

**Coverage:**
- ESS (Elastic Stack Security)
- Serverless (Complete, Essentials, Basic tiers)
- Multi-user scenarios
- License restrictions
- Context-aware features

### 2. Scout Infrastructure Implementation

#### Page Objects Created
```
kbn-scout-security/src/playwright/fixtures/test/page_objects/
├── assistant_page.ts        (~955 lines) - Main AI Assistant interactions
├── security_common.ts        (~54 lines)  - Common Security Solution helpers  
├── security_navigation.ts    (~108 lines) - Navigation patterns
└── alerts_table.ts           (enhanced)   - Alert management
```

**Features:**
- 40+ locators for UI elements
- 30+ action methods
- 15+ assertion helpers
- Context-aware operations (rule, alert, general)
- Timeout constant integration

#### API Services Created
```
kbn-scout-security/src/playwright/fixtures/worker/apis/
├── assistant.ts        (~242 lines) - Conversation & prompt APIs
├── connectors.ts       (~397 lines) - Connector management APIs
└── utils.ts            (~243 lines) - Shared utilities & helpers
```

**Capabilities:**
- Conversation CRUD operations
- Prompt management (system & quick)
- Connector lifecycle (Azure OpenAI, Bedrock, Slack)
- Proper cleanup with ES polling verification
- Space-aware operations
- Performance tracking

#### Authentication Architecture
Implemented **dual-context authentication** solving the critical issue where user-scoped resources (conversations, prompts) were invisible in the UI:

- **Worker-scoped services (`apiServices`)**: Global cleanup operations
- **Test-scoped services (`browserScopedApis`)**: User-scoped resources matching browser user
- Type-safe separation preventing authentication bugs

## 🏗️ Architecture Highlights

### API Service Pattern (Scout Innovation)

```typescript
// Centralized, organized API access
const services = {
  assistant: {
    createConversation: (body) => Promise<Conversation>,
    deleteAllConversations: () => Promise<void>,
    createPrompts: (prompts) => Promise<void>,
    // ... more methods
  },
  connectors: {
    createAzureOpenAI: () => Promise<Connector>,
    createBedrock: () => Promise<Connector>,
    deleteAll: () => Promise<void>,
    // ... more methods
  },
  // ... more services
};

// Usage in tests - clean, typed, discoverable
test('example', async ({ apiServices }) => {
  const connector = await apiServices.connectors.createAzureOpenAI();
  await apiServices.assistant.createConversation({ connectorId: connector.id });
});
```

### Timeout Constants Architecture

```typescript
// Before: Magic numbers scattered throughout
await element.waitFor({ timeout: 10000 });
await page.waitForLoadState('networkidle', { timeout: 5000 });

// After: Centralized, self-documenting
export const TIMEOUTS = {
  UI_ELEMENT_SHORT: 2000,
  UI_ELEMENT_STANDARD: 5000,
  UI_ELEMENT_LONG: 10000,
  UI_ELEMENT_EXTRA_LONG: 15000,
  NETWORK_IDLE: 5000,
} as const;

await element.waitFor({ timeout: TIMEOUTS.UI_ELEMENT_LONG });
await page.waitForLoadState('networkidle', { timeout: TIMEOUTS.NETWORK_IDLE });
```


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

enriquesanchez-elastic committed:
feat: new scout test cases for security

enriquesanchez-elastic committed:
Merge branch 'main' into security-scout-tests

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/styled_components_mapping'

dmlemeshko commented on x-pack/solutions/security/test/scout/ui/tests/security_navigation.spec.ts:
```diff
@@ -0,0 +1,70 @@
+    await pageObjects.securityNavigation.goToTimelines();
+  });
+
+  test('navigates to the Dashboards landing page', async ({ page, pageObjects }) => {
+    await pageObjects.securityNavigation.clickDashboards();
+    expect(page.url()).toContain('/app/security/dashboards');
+  });
+
+  test('navigates to the Overview page', async ({ page, pageObjects }) => {
+    await pageObjects.securityNavigation.clickOverview();
+    expect(page.url()).toContain('/app/security/overview');
+  });
```
does it really needs to be a new `test` block? For each `test` block scout creates a fresh browser context, which takes quite some time if you have 10x blocks. The alternative we suggest is to group navigation checks based on levels and use `test.step` to have a nice clear report:

```
  test.step('navigates to the Dashboards landing page', async ({ page, pageObjects }) => {
    await pageObjects.securityNavigation.clickDashboards();
    expect(page.url()).toContain('/app/security/dashboards');
  });

  test.step('navigates to the Overview page', async ({ page, pageObjects }) => {
    await pageObjects.securityNavigation.clickOverview();
    expect(page.url()).toContain('/app/security/overview');
  });

....
```

csr commented on x-pack/solutions/security/packages/kbn-scout-security/src/playwright/fixtures/test/page_objects/security_navigation.ts:
```diff
@@ -0,0 +1,98 @@
+    this.casesLink = this.page.testSubj.locator('solutionSideNavItemLink-cases');
+  }
+
+  async goToTimelines() {
```
Following Playwright's `page.goto()` casing, we've named Scout's navigation functions using the same convention. For example:

```suggestion
  async gotoTimelines() {
```

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/335672)
* Commit: fbadbc658cd6bc6f4e05c8096feda43bc67502c6

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/335672#019913ee-c3a6-41c1-89ef-e5e1bab2c54e)

### History
* :broken_heart: [Build #335658](https://buildkite.com/elastic/kibana-pull-request/builds/335658) failed 03fdf0bc47eb62f7a4d5df532ce2ae9edd029fc1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335672","number":335672,"commit":"fbadbc658cd6bc6f4e05c8096feda43bc67502c6"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335658","number":335658,"commit":"03fdf0bc47eb62f7a4d5df532ce2ae9edd029fc1"}],"number":335672}
buildkite-pr-comment-->

enriquesanchez-elastic committed:
feat: enhance security navigation tests with new locators and improved assertions

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/335989)
* Commit: 2ddf0242170ae9cc7d70c69875958587641c0606

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/335989#0199157e-0bd1-4e7c-b536-17a86de5fad7)

### History
* :broken_heart: [Build #335672](https://buildkite.com/elastic/kibana-pull-request/builds/335672) failed fbadbc658cd6bc6f4e05c8096feda43bc67502c6
* :broken_heart: [Build #335658](https://buildkite.com/elastic/kibana-pull-request/builds/335658) failed 03fdf0bc47eb62f7a4d5df532ce2ae9edd029fc1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335989","number":335989,"commit":"2ddf0242170ae9cc7d70c69875958587641c0606"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335672","number":335672,"commit":"fbadbc658cd6bc6f4e05c8096feda43bc67502c6"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335658","number":335658,"commit":"03fdf0bc47eb62f7a4d5df532ce2ae9edd029fc1"}],"number":335989}
buildkite-pr-comment-->

csr reviewed (approved):
This is looking good to me. Nice job setting up Scout tests for the Security solution. Let us know if you need any further help.

enriquesanchez-elastic committed:
Merge branch 'main' into security-scout-tests

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/styled_components_mapping'

enriquesanchez-elastic committed:
[GenAI] Add AI Assistant tests for various license tiers and features

- Implement tests for AI Assistant under Basic License to ensure users cannot access the assistant features.
- Add tests for Security Complete license to verify AI Assistant availability.
- Create tests for Security Essentials to confirm the absence of AI Assistant.
- Develop comprehensive tests for AI Assistant messages, ensuring KQL queries can be utilized in timelines.
- Introduce tests for AI Assistant prompts, including system and quick prompts, to validate their functionality and visibility.
- Implement tests for shared conversations, ensuring proper sharing functionality and access control between users.

enriquesanchez-elastic committed:
[GenAI] Refactor AI Assistant tests to use spaceTest for improved organization and clarity

- Updated test files to replace 'test' with 'spaceTest' for consistency.
- Modified API calls to utilize page.request for better session handling.
- Enhanced error handling for license start requests.
- Adjusted conversation IDs to include scoutSpace ID for unique identification.
- Cleaned up test structure by consolidating beforeEach and afterEach hooks.
- Ensured all tests maintain functionality while improving readability and maintainability.

enriquesanchez-elastic committed:
[GenAI] Update connectorSelector expectation to use containText for better UI handling

kibanamachine committed:
[CI] Auto-commit changed files from 'security: 3rd-party dependencies'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

enriquesanchez-elastic committed:
[CI] Update CODEOWNERS to include new test files for AI Assistant and Security navigation

- Added ownership for AI Assistant tests in the scout UI.
- Included new test files for alert details URL synchronization and security navigation under the relevant teams.

enriquesanchez-elastic committed:
Refactor AssistantPage and SecurityCommonPage for improved code clarity

- Updated property accessors in AssistantPage to use public getters for consistency.
- Added eslint-disable comments for specific Playwright rules to enhance code readability.
- Modified error handling in connectors and detection_rule APIs to use 'unknown' type for better type safety.
- Cleaned up retryApiCall and pollUntilAvailable functions for improved error handling and logging clarity.

enriquesanchez-elastic committed:
Refactor kbn-scout-security API utilities for improved readability

- Removed unused SecurityParallelApiServicesFixture export from index.ts.
- Simplified destructuring of options in retryApiCall and getPollConfig functions for better clarity.
- Enhanced logging messages in pollUntilAvailable function for improved debugging and error tracking.

elasticmachine committed:
Merge branch 'main' into security-scout-tests

enriquesanchez-elastic committed:
Refactor browser authentication and user profile activation in kbn-scout

- Updated documentation to clarify user profile activation process for SAML authentication.
- Removed redundant user profile activation logic from browserAuthFixture.
- Adjusted userProfilesFixture to reflect that SAML users are auto-activated by Kibana on first authenticated request.
- Cleaned up parallel and single-thread fixture files by removing unnecessary userProfilesFixture integration.

enriquesanchez-elastic committed:
Merge branch 'main' into security-scout-tests

enriquesanchez-elastic committed:
Refactor AssistantPage to enhance code consistency and readability

- Updated property accessors in AssistantPage to use public getters for improved clarity and consistency.
- Modified assertions in various methods to use 'toBeHidden()' instead of 'not.toBeVisible()' for better semantic accuracy in Playwright tests.
- Cleaned up comments and formatting for enhanced readability.

enriquesanchez-elastic committed:
Add standardized timeout constants for Scout Security tests

- Introduced TIMEOUTS constants in a new file to standardize timeout values across tests.
- Updated AssistantPage and SecurityCommonPage to utilize the new TIMEOUTS for improved consistency and maintainability.
- Refactored test files to replace hardcoded timeout values with the corresponding TIMEOUTS constants, enhancing readability and reducing magic numbers.

enriquesanchez-elastic committed:
Enhance AI Assistant Prompts test configuration

- Updated the rule configuration in prompts.spec.ts to include an 'enabled' property for better clarity.
- Introduced a dynamic 'rule_id' based on the scout space ID and timestamp to ensure unique identification of rules.

enriquesanchez-elastic committed:
Add Playwright test configurations and AI Assistant tests

- Introduced new Playwright configuration files for parallel testing in the security solution.
- Added multiple test files for the AI Assistant, covering various functionalities including conversation management, prompt handling, and navigation through security solution pages.
- Implemented tests for alert details URL synchronization and security navigation to ensure proper functionality and user experience.

enriquesanchez-elastic committed:
Merge branch 'main' into security-scout-tests

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/353639)
* Commit: 3cf730e04e79547884b8eea7845ce80d4276ee43

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/353639#019a0c41-e949-48e2-acd5-e51d8f5fe6fe)

### History
* :yellow_heart: [Build #353560](https://buildkite.com/elastic/kibana-pull-request/builds/353560) was flaky ff82e35dd17ecdae77587d832b20237e2e14b484
* :broken_heart: [Build #353535](https://buildkite.com/elastic/kibana-pull-request/builds/353535) failed 81d2f19970d1714321d9a72c1b3c8b4708458eee
* :broken_heart: [Build #353457](https://buildkite.com/elastic/kibana-pull-request/builds/353457) failed 0d031487bfb5a96533e8311c5b11b77f69964334
* :broken_heart: [Build #353420](https://buildkite.com/elastic/kibana-pull-request/builds/353420) failed 646b7093131b66770df50005c67bf3ba4e1371af
* :broken_heart: [Build #353227](https://buildkite.com/elastic/kibana-pull-request/builds/353227) failed 4e072eff3dd9348a4b447e5949fa0ec8eb9cc606
* :broken_heart: [Build #353214](https://buildkite.com/elastic/kibana-pull-request/builds/353214) failed 293478cf30065becbae10f79f6cde72cb44bed25

cc @enriquesanchez-elastic

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353639","number":353639,"commit":"3cf730e04e79547884b8eea7845ce80d4276ee43"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353560","number":353560,"commit":"ff82e35dd17ecdae77587d832b20237e2e14b484"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353535","number":353535,"commit":"81d2f19970d1714321d9a72c1b3c8b4708458eee"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353457","number":353457,"commit":"0d031487bfb5a96533e8311c5b11b77f69964334"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353420","number":353420,"commit":"646b7093131b66770df50005c67bf3ba4e1371af"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353227","number":353227,"commit":"4e072eff3dd9348a4b447e5949fa0ec8eb9cc606"}],"number":353639}
buildkite-pr-comment-->

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/353657)
* Commit: a3421f41d7ad246831df3f3fd4dedbace8880738

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/353657#019a0c58-96f9-4da4-beb3-7801e98b1352)

### History
* :broken_heart: [Build #353639](https://buildkite.com/elastic/kibana-pull-request/builds/353639) failed 3cf730e04e79547884b8eea7845ce80d4276ee43
* :yellow_heart: [Build #353560](https://buildkite.com/elastic/kibana-pull-request/builds/353560) was flaky ff82e35dd17ecdae77587d832b20237e2e14b484
* :broken_heart: [Build #353535](https://buildkite.com/elastic/kibana-pull-request/builds/353535) failed 81d2f19970d1714321d9a72c1b3c8b4708458eee
* :broken_heart: [Build #353457](https://buildkite.com/elastic/kibana-pull-request/builds/353457) failed 0d031487bfb5a96533e8311c5b11b77f69964334
* :broken_heart: [Build #353420](https://buildkite.com/elastic/kibana-pull-request/builds/353420) failed 646b7093131b66770df50005c67bf3ba4e1371af
* :broken_heart: [Build #353227](https://buildkite.com/elastic/kibana-pull-request/builds/353227) failed 4e072eff3dd9348a4b447e5949fa0ec8eb9cc606

cc @enriquesanchez-elastic

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353657","number":353657,"commit":"a3421f41d7ad246831df3f3fd4dedbace8880738"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353639","number":353639,"commit":"3cf730e04e79547884b8eea7845ce80d4276ee43"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353560","number":353560,"commit":"ff82e35dd17ecdae77587d832b20237e2e14b484"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353535","number":353535,"commit":"81d2f19970d1714321d9a72c1b3c8b4708458eee"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353457","number":353457,"commit":"0d031487bfb5a96533e8311c5b11b77f69964334"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353420","number":353420,"commit":"646b7093131b66770df50005c67bf3ba4e1371af"}],"number":353657}
buildkite-pr-comment-->

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

enriquesanchez-elastic committed:
Refactor AI Assistant tests to improve structure and consistency; add timeouts for AI responses and enhance conversation handling

enriquesanchez-elastic committed:
feat: Enhance Security Common Page with Toast Management and Assertions

- Added locators for toast messages (success, error, body, close buttons) in SecurityCommonPage.
- Implemented methods for dismissing toasts and specific toast by index.
- Added assertions for success and error toasts, including message verification.
- Introduced methods to check for the visibility of error comments and generic toasts.

feat: Create Timeline Page Object for Kibana Security Solution

- Introduced TimelinePage class with locators for main elements, query/search, tabs, data providers, and actions.
- Implemented methods for navigating, creating, saving timelines, and managing data providers and filters.
- Added assertion methods to validate timeline state, query text, and event counts.

refactor: Update AI Assistant Tests to Use Page Objects

- Refactored tests in conversations.spec.ts, messages.spec.ts, prompts.spec.ts, and shared_conversations.spec.ts to utilize page objects for navigation and element interaction.
- Improved readability and maintainability by replacing direct element selectors with page object methods.
- Ensured onboarding modals are dismissed using page object methods for consistency across tests.

enriquesanchez-elastic committed:
fix: Update test function signature to include pageObjects parameter for proper context handling

enriquesanchez-elastic committed:
feat: Enhance AI Assistant and Detection Rule APIs with improved handling and cleanup methods

- Added wait conditions and timeouts in AssistantPage for better stability during UI interactions.
- Introduced new methods in DetectionRuleApiService for deleting all alerts and cleaning up test data.
- Updated tests to utilize new cleanup methods, ensuring a clean state before and after test execution.
- Refactored alert details synchronization tests to improve reliability and context handling.

enriquesanchez-elastic committed:
Merge branch 'main' into security-scout-tests

enriquesanchez-elastic committed:
refactor: Update AI Assistant tests to improve loading state handling

- Replaced network idle wait conditions with visibility checks for the connector selector to ensure conversations are fully loaded before state validation.
- Enhanced test reliability by implementing consistent timeout handling during UI interactions.

enriquesanchez-elastic committed:
refactor: Import TIMEOUTS in AI Assistant conversations test for improved timeout handling

elasticmachine committed:
Merge branch 'main' into security-scout-tests

enriquesanchez-elastic committed:
test: Enhance alerts table handling and improve rule execution checks

- Updated `expandAlertDetailsFlyout` method to use `.first()` for filtering alerts by rule name, addressing cases with duplicate rule names.
- Added checks to ensure alerts exist before attempting to find a specific rule name, improving robustness against empty alert tables.
- Modified tests to wait for alerts table to load and for rule execution before checking for alerts, ensuring accurate test results.
- Adjusted worker count in Playwright configuration for better resource management during tests.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/357748)
* Commit: 470d67d4f353df9b3666e146fef42836aabd8fed

### Failed CI Steps
* [Scout: [ security / security_solution ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/357748#019a4a71-911a-42b6-9526-c59890a06114)
* [Scout: [ security / security_solution ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/357748#019a4a92-f1ed-44dd-ab5a-3e4b7bd6016b)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/357748#019a4a71-911a-42b6-9526-c59890a06114) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/357748/jobs/019a4a71-911a-42b6-9526-c59890a06114/artifacts/019a4a92-cad4-4f17-9512-8c48541edde8) Scout: [ security / security_solution ] plugin / serverless-security - AI Assistant Conversations - When no conversations exist but connectors do exist, show empty convo - When invoked from rules page
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/357748#019a4a92-f1ed-44dd-ab5a-3e4b7bd6016b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/357748/jobs/019a4a92-f1ed-44dd-ab5a-3e4b7bd6016b/artifacts/019a4aac-04c7-45f0-88d6-8ca4951efc8a) Scout: [ security / security_solution ] plugin / serverless-security - AI Assistant Conversations - When no conversations exist but connectors do exist, show empty convo - When invoked from rules page
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/357748#019a4a71-911a-42b6-9526-c59890a06114) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/357748/jobs/019a4a71-911a-42b6-9526-c59890a06114/artifacts/019a4a92-cad3-4e8f-b8c6-570adb4e0f35) Scout: [ security / security_solution ] plugin / stateful - AI Assistant Conversations - When no conversations exist but connectors do exist, show empty convo - When invoked from rules page

### History
* :broken_heart: [Build #357572](https://buildkite.com/elastic/kibana-pull-request/builds/357572) failed 50d8fb1a580522f4e46e8ba0c635de793145f919
* :broken_heart: [Build #357262](https://buildkite.com/elastic/kibana-pull-request/builds/357262) failed fa1bc448be35177d5fac10dc99038c170a447bb8
* :broken_heart: [Build #357240](https://buildkite.com/elastic/kibana-pull-request/builds/357240) failed 24eceb00f3f76985d90179388c1a133cc0769eea
* :broken_heart: [Build #357162](https://buildkite.com/elastic/kibana-pull-request/builds/357162) failed 0f4a9daf8684060b212e071ab805da8060893f33
* :broken_heart: [Build #354127](https://buildkite.com/elastic/kibana-pull-request/builds/354127) failed 4bb3b621faf34936c9c5dd558562f1cc8958e671

cc @enriquesanchez-elastic

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357748","number":357748,"commit":"470d67d4f353df9b3666e146fef42836aabd8fed"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357572","number":357572,"commit":"50d8fb1a580522f4e46e8ba0c635de793145f919"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357262","number":357262,"commit":"fa1bc448be35177d5fac10dc99038c170a447bb8"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357240","number":357240,"commit":"24eceb00f3f76985d90179388c1a133cc0769eea"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357162","number":357162,"commit":"0f4a9daf8684060b212e071ab805da8060893f33"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354127","number":354127,"commit":"4bb3b621faf34936c9c5dd558562f1cc8958e671"}],"number":357748}
buildkite-pr-comment-->

enriquesanchez-elastic committed:
feat: Add onboarding tour dismissal to AssistantPage and ensure clean state in tests

- Implemented `dismissOnboardingTour` method in `AssistantPage` to handle the onboarding tour modal for first-time users, preventing it from blocking interactions.
- Updated `openFromRule` method to call `dismissOnboardingTour` before proceeding.
- Modified tests in `conversations.spec.ts` to close the assistant after each test, ensuring a clean state for subsequent tests.

enriquesanchez-elastic committed:
Refactor AI Assistant Page Object Methods

- Updated imports in index.ts to reflect new file structure for AssistantPage.
- Changed method calls in conversations.spec.ts to use new structure for assertions and connectors.
- Updated expect statements to assertions in various test files for consistency.
- Refactored method calls related to messaging, prompts, and conversations across multiple test files.
- Ensured all references to assistantPage methods are updated to match the new organization of page object methods.

enriquesanchez-elastic committed:
feat: Implement Rules Management Page Object and related actions

- Added RuleActions class for CRUD operations on rules.
- Introduced RuleSelectionActions for selecting rules via checkbox and name.
- Created RulesManagementAssertions for verifying rules state and behavior.
- Developed RulesManagementLocators for element selectors in Rules Management.
- Implemented TableOperations for handling rules table interactions.
- Established SearchFilterActions for searching and filtering rules.
- Refactored RulesManagementPage to orchestrate actions and assertions.
- Removed legacy RulesManagementPage implementation.
- Updated assistant cleanup service to handle version conflicts.
- Adjusted tests to utilize new page object structure for rules management.

enriquesanchez-elastic committed:
refactor: Simplify constructor syntax in AssistantAssertions and ConnectorActions classes; improve readability

enriquesanchez-elastic committed:
refactor: Restructure Timeline Page Object and related actions

- Refactored Timeline page object to utilize a composition pattern with specialized action classes for navigation, queries, filters, and data providers.
- Introduced new classes for handling timeline actions, including NavigationActions, QueryActions, DataProviderActions, and FilterActions.
- Created TimelineLocators for centralized access to UI element locators.
- Updated TimelineAssertions for improved state verification.
- Removed the legacy TimelinePage implementation and replaced it with a modular structure for better maintainability and readability.
- Adjusted tests to align with the new page object structure.

enriquesanchez-elastic committed:
feat: Add basic license alert assignment tests

- Introduced a new test suite for verifying alert user assignment functionality under a basic license.
- Implemented setup and teardown processes to manage test data and rule creation.
- Added assertions to ensure that users with a basic license cannot update alert assignees, including checks for hidden menu items and disabled buttons.
- Enhanced test reliability by ensuring proper cleanup of alerts and rules after each test execution.

enriquesanchez-elastic committed:
Merge branch 'main' into security-scout-tests

enriquesanchez-elastic committed:
Remove duplicate entry for "@kbn/react-query" in tsconfig.json

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/361763)
* Commit: 80293a39c52fa2e7c5975107df698701334f7e29

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/361763#019a9151-cf79-444c-9d5d-63a3930cd649)

### History
* :broken_heart: [Build #358867](https://buildkite.com/elastic/kibana-pull-request/builds/358867) failed bd379bfdabca67d81f99d2dec7aa461cd581a65a
* :broken_heart: [Build #357748](https://buildkite.com/elastic/kibana-pull-request/builds/357748) failed 470d67d4f353df9b3666e146fef42836aabd8fed
* :broken_heart: [Build #357572](https://buildkite.com/elastic/kibana-pull-request/builds/357572) failed 50d8fb1a580522f4e46e8ba0c635de793145f919
* :broken_heart: [Build #357262](https://buildkite.com/elastic/kibana-pull-request/builds/357262) failed fa1bc448be35177d5fac10dc99038c170a447bb8
* :broken_heart: [Build #357240](https://buildkite.com/elastic/kibana-pull-request/builds/357240) failed 24eceb00f3f76985d90179388c1a133cc0769eea

cc @enriquesanchez-elastic

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361763","number":361763,"commit":"80293a39c52fa2e7c5975107df698701334f7e29"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358867","number":358867,"commit":"bd379bfdabca67d81f99d2dec7aa461cd581a65a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357748","number":357748,"commit":"470d67d4f353df9b3666e146fef42836aabd8fed"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357572","number":357572,"commit":"50d8fb1a580522f4e46e8ba0c635de793145f919"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357262","number":357262,"commit":"fa1bc448be35177d5fac10dc99038c170a447bb8"}],"number":361763}
buildkite-pr-comment-->

enriquesanchez-elastic committed:
Remove alert assignments test for basic license from security_solution

enriquesanchez-elastic committed:
Refactor error handling in connector actions and assistant cleanup service

- Updated catch blocks to omit error variable, simplifying error handling logic.
- Ensured that the code continues execution without interruption if an error occurs, particularly when waiting for responses or checking index existence.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/362206)
* Commit: 9cf62ed65763a321a245c3677e510c1b433761c8

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/362206#019a964d-fcf8-4b76-8ae9-b5a9887fdb5e)

### History
* :broken_heart: [Build #361936](https://buildkite.com/elastic/kibana-pull-request/builds/361936) failed 006f25ec4adc1d9d8c5586d2532bfd21bb6b7fe9
* :broken_heart: [Build #361763](https://buildkite.com/elastic/kibana-pull-request/builds/361763) failed 80293a39c52fa2e7c5975107df698701334f7e29
* :broken_heart: [Build #358867](https://buildkite.com/elastic/kibana-pull-request/builds/358867) failed bd379bfdabca67d81f99d2dec7aa461cd581a65a
* :broken_heart: [Build #357748](https://buildkite.com/elastic/kibana-pull-request/builds/357748) failed 470d67d4f353df9b3666e146fef42836aabd8fed
* :broken_heart: [Build #357572](https://buildkite.com/elastic/kibana-pull-request/builds/357572) failed 50d8fb1a580522f4e46e8ba0c635de793145f919
* :broken_heart: [Build #357262](https://buildkite.com/elastic/kibana-pull-request/builds/357262) failed fa1bc448be35177d5fac10dc99038c170a447bb8

cc @enriquesanchez-elastic

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/362206","number":362206,"commit":"9cf62ed65763a321a245c3677e510c1b433761c8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361936","number":361936,"commit":"006f25ec4adc1d9d8c5586d2532bfd21bb6b7fe9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361763","number":361763,"commit":"80293a39c52fa2e7c5975107df698701334f7e29"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358867","number":358867,"commit":"bd379bfdabca67d81f99d2dec7aa461cd581a65a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357748","number":357748,"commit":"470d67d4f353df9b3666e146fef42836aabd8fed"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357572","number":357572,"commit":"50d8fb1a580522f4e46e8ba0c635de793145f919"}],"number":362206}
buildkite-pr-comment-->

enriquesanchez-elastic committed:
Refactor AI Assistant conversation tests to improve variable naming

- Updated variable names in the `beforeEach` hooks and test functions for clarity, replacing `scoutSpace` with `_scoutSpace` and `page` with `_page`.
- This change enhances code readability and maintains consistency across test cases.

enriquesanchez-elastic committed:
Merge branch 'main' into security-scout-tests

kibanamachine committed:
Changes from node scripts/lint_ts_projects --fix
